pkg/cmd/server/crypto/crypto.go

```diff
@@ -542,20 +558,37 @@ func CertsFromPEM(pemCerts []byte) ([]*x509.Certificate, error) {
 }
 
 // Can be used as a certificate in http.Transport TLSClientConfig
-func newClientCertificateTemplate(subject pkix.Name) (*x509.Certificate, error) {
+func newClientCertificateTemplate(subject pkix.Name, expireDays int, currentTime func() time.Time) *x509.Certificate {
+	var lifetimeInDays = DefaultCertificateLifetimeInDays
+	if expireDays > 0 {
+		lifetimeInDays = expireDays
+	}
+
+	if lifetimeInDays > DefaultCertificateLifetimeInDays {
+		warnAboutCertificateLifeTime(subject.CommonName, DefaultCertificateLifetimeInDays)
+	}
+
+	lifetime := time.Duration(lifetimeInDays) * 24 * time.Hour
+
 	return &x509.Certificate{
 		Subject: subject,
 
 		SignatureAlgorithm: x509.SHA256WithRSA,
 
-		NotBefore:    time.Now().Add(-1 * time.Second),
-		NotAfter:     time.Now().Add(lifetime),
+		NotBefore:    currentTime().Add(-1 * time.Second),
+		NotAfter:     currentTime().Add(lifetime),
 		SerialNumber: big.NewInt(1),
 
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 		BasicConstraintsValid: true,
-	}, nil
+	}
+}
+
+func warnAboutCertificateLifeTime(name string, defaultLifetimeInDays int) {
+	defaultLifetimeInYears := defaultLifetimeInDays / 365
```

Comment:
I still think that there is no problem at all. 

First of all, this code calculates number of years that will be shown in the warning and nothing more, so, the biggest bug that can occur here is a misleading message. 

Second, is that this code do a conversion from a user-provided number of days to number of years. It means that user should take care about leap years when he is calculation number of days. Let's use your example: user specified 731 (2 years and 1 day). In this case calculation will end up in 2 years (731/365=2). If user specified 730 (2 years) then result will be 2 years too.

@soltysh What I missed here?